whisper_medusa/dataset/dataset.py

```diff
@@ -0,0 +1,171 @@
+import logging
+from dataclasses import dataclass
+from typing import Dict, List, Union
+import pandas as pd
+import torch
+import torchaudio
+from transformers import WhisperProcessor
+from whisper_medusa.utils.utils import token_padding
+
+SAMPLE_RATE = 16_000
+
+
+class ASRDataSet(torch.utils.data.Dataset):
+    """
+    ASR Dataset
+
+    Parameters
+    ----------
+    data_path : str. Path to csv file with columns: "sentence", "path".
+    """
+
+    def __init__(
+        self,
+        data_path,
+        split,
+        processor: WhisperProcessor,
+        target_sample=16_000,
+        debug_mode=False,
+    ):
+        assert split in [
+            "train",
+            "val",
+            "test",
+        ]  # sanity for case we will use split later.
+        self.split = split
+        self.debug_mode = debug_mode
+
+        # read csv file + create dataset
+        self.data_path = data_path
+        self.dataset_df = pd.DataFrame()
+        self._init_dataset_obj()
+
+        self.dataset = self.dataset_df.to_dict("records")
+        self.length = len(self.dataset)
+
+        self.target_sample_rate = target_sample
+
+        self.processor = processor
+
+    def _init_dataset_obj(self):
+        self.dataset_df = pd.read_csv(self.data_path)
+        self.dataset_df.sentence = self.dataset_df.sentence.fillna("")
+
+
+        # in case of debug mode, sample only 1000 samples
+        if self.debug_mode and self.split in ["val", "test"]:
+            sample_size = min(1000, len(self.dataset_df))
```

Comment:
Let's add the number of debug examples as an argument in the parser and then pass it to the dataset object.

Reply:
add the "debug-examples" parameters